src/library_syscall.js

```diff
@@ -453,7 +453,7 @@ var SyscallsLibrary = {
         if (!stream.tty) return -{{{ cDefine('ENOTTY') }}};
         return 0;
       }
-      default: abort('bad ioctl syscall ' + op);
+      default: return -{{{ cDefine('ENOSYS') }}}; // bad ioctl syscall
```

Comment:
According the man page I think this should be `EINVAL`  (request or argp is not valid).

Also, can you add a test for this?  (maybe add `test/unistd/ioctl.c`)